### ISSUE TITLE
Test Accuracy

### ISSUE THREAD

**Issue by ag1988** [HazyResearch/state-spaces#14]
Hello,

Sorry for another silly question. While training on s4-lra-cifar (on A100) I get the following final log:

```
Epoch 94, global step 85499: val/accuracy was not in top 1
Epoch 96:  80% 966/1200 [01:16<00:18, 12.63it/s, loss=0.144, v_num=xyj4, val/accuracy=0.866, val/loss=0.527, test/accuracy=0.865, test/loss=0.533, train/accuracy=0.946, train/loss=0.151]
Epoch 95, global step 86399: val/accuracy reached 0.86580 (best 0.86580), saving model to "blah/checkpoints/val/accuracy.ckpt" as top 1
Epoch 97:  78% 938/1200 [01:15<00:21, 12.38it/s, loss=0.142, v_num=xyj4, val/accuracy=0.864, val/loss=0.536, test/accuracy=0.864, test/loss=0.539, train/accuracy=0.945, train/loss=0.156]
Epoch 96, global step 87299: val/accuracy was not in top 1
Epoch 98:  75% 900/1200 [01:14<00:24, 12.06it/s, loss=0.143, v_num=xyj4, val/accuracy=0.866, val/loss=0.529, test/accuracy=0.865, test/loss=0.535, train/accuracy=0.945, train/loss=0.153]
Epoch 97, global step 88199: val/accuracy was not in top 1
Epoch 99:  79% 945/1200 [01:15<00:20, 12.45it/s, loss=0.137, v_num=xyj4, val/accuracy=0.863, val/loss=0.534, test/accuracy=0.864, test/loss=0.539, train/accuracy=0.946, train/loss=0.151]
Epoch 98, global step 89099: val/accuracy was not in top 1

Epoch 99, global step 89999: val/accuracy was not in top 1
Saving latest checkpoint...
```

1. Does this mean that the test accuracy measured at the checkpoint with the best validation accuracy is 86.5? 
2. If no, what command should I use to measure the test accuracy that you report in the paper? (I am assuming that the accuracy you report in the paper is indeed the test acc at the checkpoint with best val acc) 
3. You report 87.26 on lra-cifar - is this normal to get this gap? I hope I'm not making a mistake in interpreting the metrics. 

I would be grateful if you could help with these questions and apologize in advance if they seem too basic.

Thank you again for sharing you wonderful repo,
Ankit

**Comment by albertfgu** [comment]
Yes, the best validation accuracy was at epoch 95 with 86.5% validation and test accuracy.

A gap like this to the reported 87% is quite normal due to noise. Actually the gap here is probably due to the fact that the model changed very slightly during the SaShiMi code release. You can probably get 87% or even higher with a bit of hyperparameter tuning. I would recommend passing in `model.layer.postact=glu` `model.layer.bidirection=true` to start, which should boost performance.

**Comment by ag1988** [comment]
Thank you very much Albert for clarifying this so promptly!

Regards,
Ankit
